xbmc/pvr/windows/GUIViewStatePVR.cpp

```diff
@@ -77,8 +77,11 @@ void CGUIViewStateWindowPVRGuide::SaveViewState(void)
 
 CGUIViewStateWindowPVRTimers::CGUIViewStateWindowPVRTimers(const int windowId, const CFileItemList& items) : CGUIViewStatePVR(windowId, items)
 {
-  AddSortMethod(SortByLabel, 551, LABEL_MASKS("%L", "%I", "%L", ""));   // FileName, Size | Foldername, empty
-  AddSortMethod(SortByDate, 552, LABEL_MASKS("%L", "%J", "%L", "%J"));  // FileName, Date | Foldername, Date
+  SortAttribute sortAttributes(SortAttributeIgnoreFolders);
+  if (CSettings::Get().GetBool("filelists.ignorethewhensorting"))
+    sortAttributes = (SortAttribute) (sortAttributes | SortAttributeIgnoreArticle);
```

Comment:
This all seems to be because SortAttribute is defined in SortUtils.h as an enumeration which doesn't consider both 'IgnoreFolders' and 'IgnoreArticle', although the sort code appears to cope with it and there seem to be other places which use them in combination.
``` cpp
typedef enum {
  SortAttributeNone           = 0x0,
  SortAttributeIgnoreArticle  = 0x1,
  SortAttributeIgnoreFolders  = 0x2
} SortAttribute;
```
maybe @Montellese has a suggestion? (His name seems to be on the commit which first introduced SortUtils.h)

I figure any re-work to adjust this ought to be a separate PR as it could have wider ranging implications.
For now I'll update this to use the same syntax as in FileItem.cpp:
```cpp
xbmc/FileItem.cpp:1945:    sortDescription.sortAttributes = (SortAttribute)((int)sortDescription.sortAttributes | SortAttributeIgnoreFolders);
xbmc/FileItem.cpp:1953:    sortDescription.sortAttributes = (SortAttribute)((int)sortDescription.sortAttributes | SortAttributeIgnoreFolders);
```
I'm sure you'll let me know if something more is required.